### PR TITLE
feat(server): add compare different url

### PR DESCRIPTION
**Pull request by pndllxzzy**

### CODE REVIEW

**Comment by pndllxzzy** [comment]
@googlebot I signed it!

**Comment by patrickhulce** [comment]
thanks for the PR @pndllxzzy! It gets a little crowded by sticking it up there though and the click target moves when you hover because of "Open Report" :/
image


Other ideas:

 - hide the base and compare URL selectors behind the menu where you select the build/hash
 - have both dropdowns in the original location of the compare URL dropdown (add more labels to differentiate)

**Comment by pndllxzzy** [comment]
Thanks for your ideas.Do you mean like this?
image

**Comment by patrickhulce** [comment]
Yeah, that seems to get the job done even if it's not the prettiest. WDYT?

**Comment by pndllxzzy** [comment]
@patrickhulce I agree with you. I'm not good at beautiful layout. If you have any ideas, please tell me.🤝🏻

File: packages/server/src/ui/routes/build-view/lhr-comparison.jsx [hunk 1]
```diff
@@ -113,13 +113,14 @@ export function computeAuditGroups(lhr, baseLhr, options) {
   return auditGroups;
 }
 
-/** @param {{hookElements: LHCI.HookElements<'dropdowns'>, selectedAuditId?: string | null, lhr?: LH.Result, baseLhr?: LH.Result, percentAbsoluteDeltaThreshold: number, setPercentAbsoluteDeltaThreshold: (x: number) => void}} props */
```
Comment (patrickhulce) [inline packages/server/src/ui/routes/build-view/lhr-comparison.jsx:116]: I think we can leave this just as `dropdowns` it was already set to be plural for more than one. 

use a fragment when it's used instead?

File: packages/server/src/ui/routes/build-view/build-view.jsx [hunk 2]
```diff
@@ -126,7 +130,7 @@ const BuildView_ = props => {
             lhr={lhr}
             isDimmed={openBuildHash === 'base'}
             isOpen={openBuildHash === 'compare'}
-            setLhrLinkHover={setLhrLinkHover}
+            setLhrLinkHover={setLhrCompareLinkHover}
```
Comment (patrickhulce) [inline packages/server/src/ui/routes/build-view/build-view.jsx:133]: nice split 👍

File: packages/server/src/ui/routes/build-view/build-view.jsx [hunk 3]
```diff
@@ -159,24 +163,42 @@ const BuildView_ = props => {
...
         hookElements={{
           warnings: computeWarnings(warningProps).hasWarning ? (
             <BuildViewWarnings {...warningProps} />
           ) : (
             undefined
           ),
-          dropdowns: (
+          dropdownsBase: (
+            <Dropdown
+              label="BaseUrl"
```
Comment (patrickhulce) [inline packages/server/src/ui/routes/build-view/build-view.jsx:178]:
```suggestion
              label="Base URL"
```

File: packages/server/src/ui/routes/build-view/build-view.jsx [hunk 4]
```diff
@@ -159,24 +163,42 @@ const BuildView_ = props => {
...
+              }}
+              options={availableUrlOptions}
+            />
+          ),
+          dropdownsCompare: (
             <Dropdown
-              label="URL"
-              className="dropdown--url"
-              value={selectedUrl}
+              label="CompareUrl"
```
Comment (patrickhulce) [inline packages/server/src/ui/routes/build-view/build-view.jsx:192]:
```suggestion
              label="Compare URL"
```

**Review by patrickhulce — COMMENTED** [review]
thanks @pndllxzzy this is great! I think you might need to update some snapshot tests as well?

File: packages/server/src/ui/routes/build-view/lhr-comparison.jsx (hunk 1 above)
Comment (pndllxzzy) [reply packages/server/src/ui/routes/build-view/lhr-comparison.jsx:116]: Do you mean like this?
```
<LhrComparison
  ...
  hookElements={{
    ...
    dropdowns: (
      <Fragment>
        <Dropdown
          label="Base Url"
          className="dropdown--url dropdown--base-url"
          value={baseUrl}
          setValue={url => {
            const to = new URL(window.location.href);
            to.searchParams.set('baseUrl', url);
            to.searchParams.set('compareUrl', compareUrl);
            route(`${to.pathname}${to.search}`);
          }}
          options={availableUrlOptions}
        />
        <Dropdown
          label="Compare Url"
          className="dropdown--url dropdown--compare-url"
          value={compareUrl}
          setValue={url => {
            const to = new URL(window.location.href);
            to.searchParams.set('baseUrl', baseUrl);
            to.searchParams.set('compareUrl', url);
            route(`${to.pathname}${to.search}`);
          }}
          options={availableUrlOptions}
        />
      </Fragment>
    ),
  }}
/>
```

**Comment by pndllxzzy** [comment]
@patrickhulce How to update snapshot?
Use this command?
`jest --updateSnapshot`

**Comment by patrickhulce** [comment]
Yep! I tend to use `-u` but that works too :)

**Review by patrickhulce — CHANGES_REQUESTED** [review]
just these two leftover from previous review but other than that looks good with test fixes! :) thanks for doing this @pndllxzzy

**Comment by pndllxzzy** [comment]
@patrickhulce Should I rebase all commits to one?

**Comment by patrickhulce** [comment]
> Should I rebase all commits to one?

Don't worry about it we'll squash when we merge :) I'll take a closer look here tomorrow 👍

**Review by patrickhulce — APPROVED** [review]
thanks for sticking with this @pndllxzzy! 🎉

**Pull request merged**